motif_inference/module_wraper.py

```diff
@@ -1,636 +1,718 @@
+    cutoffs_bcs = []
+    for bc in biological_conditions:
+        bc_folder = os.path.join(motif_inference_output_path, bc)
+        bc_memes_folder = os.path.join(bc_folder, 'memes')
+        bc_cutoffs_folder = os.path.join(bc_folder, 'cutoffs')
+        os.makedirs(bc_cutoffs_folder, exist_ok=True)
+        for file_name in sorted(os.listdir(bc_memes_folder)):
+            meme_path = os.path.join(bc_memes_folder, file_name)
+            output_path = os.path.join(bc_cutoffs_folder, file_name)
+            done_path = f'{logs_dir}/14_{bc}_{file_name}_done_compute_cutoffs.txt'
+            if not os.path.exists(done_path):
+                all_cmds_params.append([meme_path, output_path, done_path, '--total_memes', memes_per_bc[bc]])
+                cutoffs_bcs.append(bc)
+            else:
+                logger.debug(f'Skipping calculate cutoff as {done_path} exists')
+                num_of_expected_results += 1
+
+    if len(all_cmds_params) > 0:
+        for cmds_params, bc in zip(all_cmds_params, cutoffs_bcs):
+            cmd = submit_pipeline_step(f'{src_dir}/motif_inference/{script_name}',
+                                [cmds_params],
+                                logs_dir, f'{bc}_cutoffs',
+                                queue_name, verbose)
+            num_of_expected_results += 1  # a single job for each biological condition
+
+        wait_for_results(script_name, logs_dir, num_of_expected_results, example_cmd=cmd,
+                        error_file_path=error_path, suffix='_done_compute_cutoffs.txt')
+    else:
+        logger.info('Skipping calculate cutoffs, all exists')
+
+
+def process_params(args, multi_exp_config_inference, argv):
+    done_file = args.done_file_path
+    base_map =  args.__dict__
+    keys = base_map.keys()
+    base_map = change_key_name(base_map, map_names_command_line)
+    if multi_exp_config_inference:    
+        f = open(multi_exp_config_inference)
+        multi_experiments_dict = json.load(f)
+        # validation of the json file
+        is_valid = is_valid_json_structure(multi_exp_config_inference, multi_experiments_dict, schema_inference, logger)
+        if not is_valid:
+            return 
+        configuration = multi_experiments_dict['configuration']
+        base_map.update(configuration)
+        runs = multi_experiments_dict['runs']
+        for run in runs:
+            dict_params = base_map.copy()
+            dict_params.update(runs[run])
+            argv_new = []
+            argv_new.append(argv[0])
+            for k in keys:
+                val = str(dict_params[map_names_command_line[k]])
+                if (val != 'None') and (val != 'False'):
+                    argv_new.append(k)
+                    argv_new.append(val)
+            # create new list of argv of the specific run.
+            call_infer_motifs(dict_params, run, argv_new)
```

Comment:
The function is blocking, i.e. each experiment waits for previous experiment to complete.
Don't we want to run the experiments in parallel?